pkg/cmd/docgen/diagrams_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/docgen/extract"
+	"github.com/stretchr/testify/require"
+)
+
+// TestStmtSpecs ensures that each statement in sql.y identified by
+// %type <tree.Statement> has a corresponding entry in the stmtSpec slice
+// returned by getAllStmtSpecs().
+// Statements that are specified to be skipped, unimplemented or have no
+// branches should be ignored.
+func TestStmtSpecs(t *testing.T) {
+	sqlGrammarFile := "../../sql/parser/sql.y"
+	// runBNF with a local file does not use the API.
+	bnfAPITimeout := time.Duration(0)
+
+	file, err := os.Open(sqlGrammarFile)
+	require.NoError(t, err)
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	bnf, err := runBNF(sqlGrammarFile, bnfAPITimeout)
+	require.NoError(t, err)
+
+	br := func() io.Reader {
+		return bytes.NewReader(bnf)
+	}
+
+	grammar, err := extract.ParseGrammar(br())
+	require.NoError(t, err)
+
+	sqlStmts := make(map[string]bool)
+
+	stmtRegex, err := regexp.Compile(`%type <tree.Statement>`)
```

Comment:
feels like duplicated logic, can we have a function that extracts all statements from a file and/or run `getAllStmtSpecs` then check each file exists?

also i think will be caught by the linter, `make generate` runs as part of the linter and if the files generated don't match it will error. Thus, I would suggest nixxing this test.